File: sdks/python/examples/bulk_operations/replay.py
# > Setup

from datetime import datetime, timedelta

from hatchet_sdk import BulkCancelReplayOpts, Hatchet, RunFilter, V1TaskStatus

hatchet = Hatchet()

workflows = hatchet.workflows.list()

assert workflows.rows

workflow = workflows.rows[0]

# !!

# > List runs
workflow_runs = hatchet.runs.list(workflow_ids=[workflow.metadata.id])
# !!

# > Replay by run ids
workflow_run_ids = [workflow_run.metadata.id for workflow_run in workflow_runs.rows]

bulk_replay_by_ids = BulkCancelReplayOpts(ids=workflow_run_ids)

hatchet.runs.bulk_replay(bulk_replay_by_ids)
# !!

# > Replay by filters
bulk_replay_by_filters = BulkCancelReplayOpts(
    filters=RunFilter(
        since=datetime.today() - timedelta(days=1),
        until=datetime.now(),
        statuses=[V1TaskStatus.RUNNING],
        workflow_ids=[workflow.metadata.id],
        additional_metadata={"key": "value"},
    )
)

hatchet.runs.bulk_replay(bulk_replay_by_filters)
# !!

File: examples/ruby/bulk_operations/replay.rb
# frozen_string_literal: true

require "hatchet-sdk"

# > Setup
hatchet = Hatchet::Client.new

workflows = hatchet.workflows.list

workflow = workflows.rows.first

# > List runs
workflow_runs = hatchet.runs.list(workflow_ids: [workflow.metadata.id])

# > Replay by run ids
workflow_run_ids = workflow_runs.rows.map { |run| run.metadata.id }

hatchet.runs.bulk_replay(ids: workflow_run_ids)

# > Replay by filters
hatchet.runs.bulk_replay(
  since: Time.now - 86_400,
  until_time: Time.now,
  statuses: ["FAILED"],
  workflow_ids: [workflow.metadata.id],
  additional_metadata: { "key" => "value" }
)
